uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/Service/LeaveRequest.php

```diff
@@ -130,6 +128,27 @@ protected function update($params) {
     return $this->createRequestWithBalanceChanges($params);
   }
 
+  /**
+   * Return an error message
+   * either for specific invalid status transition cases
+   * or a default generic message
+   *
+   * @param array $params
+   *
+   * @return string
+   */
+  private function getErrorMessageForInvalidStatusTransition($params) {
+    $leaveStatuses = LeaveRequest::getStatuses();
+    $isOwnRequest = CRM_Core_Session::getLoggedInContactID() === $params['contact_id'];
+
+    if ($isOwnRequest && (int)$params['status_id'] === (int)$leaveStatuses['approved']) {
+      return "You can't approve your own leave requests";
+    }
+
+    return "You can't change the Leave Request status from " .
+      $this->getCurrentStatus($params) . " to {$params['status_id']}";
```

Comment:
Would be a good idea to fix this error message so that instead of showing the status id's it shows the actual status label

Reply:
Yes, that was the intention, but then I realised there is no code that stores the actual labels. Instead, we have strings such as `awaiting_approval`. So I wrote a function that transforms it into a human readable string:

```php
ucwords(str_replace('_', ' ', array_search($leaveStatusId, $leaveStatuses)));
// will transform "awaiting_approval" to "Awaiting Approval"
```